docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -44,136 +48,282 @@ To understand some of the examples below, let's imagine the following simple wor
+The following example shows how to define named inputs for the entire workspace in `nx.json`.
 
-The following definition includes all files of the project itself as well as all of its dependencies in the cache hash, hence telling Nx to invalidate the cache whenever
+```jsonc {% fileName="nx.json" highlightLines=["3-6"] %}
+{
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*"] // Default Inputs
+  }
+}
+```
 
-- any file of the project itself changes
-- any file of any dependency changes
+{% /tab %}
+{% tab label="Project Level (project.json)" %}
 
-```jsonc {% fileName="nx.json" %}
+```jsonc {% fileName="nx.json" highlightLines=["2-6"] %}
 {
   "namedInputs": {
-    "default": ["{projectRoot}/**/*"]
+    "default": ["{projectRoot}/**/*", "sharedGlobals"], // Default Inputs
+    "production": ["default", "!{projectRoot}/some-dev-only-files"], // Production Inputs
+    "sharedGlobals": [] // Shared Global Inputs
   },
   "targetDefaults": {
-    "build": {
-      "inputs": ["default", "^default"]
-    }
+    "inputs": ["default", "^default"]
   }
 }
 ```
 
-This definition is the default behavior of Nx, even if you don't specify any `inputs` at all. This is a rather cautious approach. It will always give you correct output, but it might re-run a task in some cases where the cache could have been used instead.
+{% /tab %}
+{% tab %}
+Naming a set of inputs with the same name as a set of inputs for the workspace will override the meaning of that set for the project.
+{% /tab %}
+{% /tabs %}
 
-Note, Nx uses the [minimatch](https://github.com/isaacs/minimatch) library to process glob patterns.
+### Default Inputs
 
-{% callout title="Replacements" %}
+The `default` inputs include all files in a project as well as any shared global inputs which should be considered for all tasks.
 
-Note how you can use `{projectRoot}` and `{workspaceRoot}` as placeholders to simplify writing glob definitions.
+### Shared Global Inputs
 
-{% /callout %}
+The `sharedGlobal` inputs include things that Nx should always look at when determining computation hashes. For instance, this could be the OS where the command is being run or the version of Node.
+
+### Production Inputs
 
-If you're wondering what `namedInputs` are, read the next section.
+The `production` inputs only include the set of files in a project which will affect behavior of the project in production.
+For instance, the `main.ts` of an application will be compiled as the main application logic and will affect the end user experience while files such as `.eslintrc.json` only affect the tools used by the developers and have no direct impact on the end user experience.
+In general, it is best to define the `production` inputs as the `default` inputs (everything) excluding the specific files which are known not to affect end user experience. This makes it so that by default, all files are considered to affect end user behavior unless excluded from the `production` fileset.
```

Comment:
I think this needs to be clear that its just explaining the default generated named inputs. I know we mention it in the first tab here:
<img width="932" alt="image" src="https://github.com/nrwl/nx/assets/4332460/370677fe-75b0-40cc-ac2d-45f08fc35e60">
but as soon as you switch tabs, you lose that context. I got confused because I saw this:
<img width="969" alt="image" src="https://github.com/nrwl/nx/assets/4332460/bbede1b7-e20c-4207-a357-7b75b259ccc1">

The headings of each of the "namedInput" names make it seem more important that it is.